### PR TITLE
Graphql server websocket & reactive - internal review!

### PR DESCRIPTION
**Pull request by DxCx**
Hidden PR for internal review porpuse

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "lerna": "2.0.0-beta.35",
-  "version": "0.5.1",
+  "version": "0.6.0",
```
Comment (helfer) [inline lerna.json:3]: I think you have to undo this so we could release with lerna again, right?

File: packages/graphql-server-reactive-core/src/RequestsManager.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+          }
+
+          this._unsubscribeAll();
+        };
+      });
+
+      this.executeReactive = graphqlOptions.executor.executeReactive.bind(graphqlOptions.executor);
+  }
+
+  public get responds(): IObservable<RGQLPacket> {
```
Comment (helfer) [inline packages/graphql-server-reactive-core/src/RequestsManager.ts]: should be "response" (also in other places)

File: packages/graphql-server-reactive-core/src/runQueryReactive.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    validate,
+    formatError,
+    specifiedRules,
+} from 'graphql';
+import { LogAction, LogFunction, LogMessage, LogStep, QueryOptions} from 'graphql-server-core';
+import { Observable, IObservable } from './Observable';
+import { RGQLExecuteFunction } from './reactiveOptions';
+
+export interface ReactiveQueryOptions extends QueryOptions {
+  executeReactive: RGQLExecuteFunction;
```
Comment (helfer) [inline packages/graphql-server-reactive-core/src/runQueryReactive.ts]: I would just call this argument `executor`

File: packages/graphql-server-reactive-core/src/messageTypes.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { ExecutionResult } from 'graphql';
+
+// Refer to https://github.com/apollographql/graphql-server/issues/272#issuecomment-278805955 for more information about protocol
```
Comment (helfer) [inline packages/graphql-server-reactive-core/src/messageTypes.ts]: I think we need to discuss the transport and message types with Uri and the folks working on GraphQL subscriptions. It makes a whole lot of sense to have a shared transport, so they probably have their opinions about the message types as well.

File: packages/graphql-server-reactive-core/src/RequestsManager.ts [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+  RGQL_MSG_DATA,
+  RGQL_MSG_START,
+  RGQL_MSG_STOP,
+  RGQL_MSG_KEEPALIVE,
+  RGQLPacket,
+  RGQLPacketData,
+  RGQLPayloadStart,
+} from './messageTypes';
+
+export class RequestsManager {
```
Comment (helfer) [inline packages/graphql-server-reactive-core/src/RequestsManager.ts]: Personally I would prefer to do all the connection handling in the ws package. The reactive core should only provide an observable runQuery function that the ws package can call when appropriate. Kind of like how subscriptions-transport-ws and graphql-subscriptions play together.

**Review by helfer — COMMENTED** [review]
I think this really nice. I like that it's split into reactive-core and ws packages already, but I think some things aren't in the right place yet. the ws package should handle everything that has to do with the transport, and the reactive-core package should just be the runQueryObservable function (or whatever we want to call it in the end). The executor should be a separate package, so it's easy to copy and write your own executor if you have different requirements. ws calls executor, which then calls reactive core or core, depending on the requirements. It could also call graphql-subscriptions. Does that make sense?

I think you should try to get feedback from other people as well, most importantly Uri.

**Comment by DxCx** [comment]
@Urigo please review as well :)

**Comment by DxCx** [comment]
@helfer, for some reason i don't have per comment reply.
i'll try to answer all at once:
1. regarding the splitting, at the beging RequestsManager and types system was part of the WS package,
but then I Realized that this implementation will fit to any transport that gives you connection and there is no reason to write a different protocol for each.
also, i do plan to release `uws` support as well right after this patch will be merged so it will be thin as well.
2. regarding lerna versioning, looks like the last release was either not released with lerna or the releasing commit was not pushed to master as 0.6.0 is out but in lerna config latest released was 0.5.1,
if we are not in sync, lerna will download the package from NPM rather then link internally,
3. the idea behind `executeReactive` is that it has the same signature as graphql-js `execute` but it returns an observable, that's why i called it that way.
so basiclly, `executor` object should have `executeReactive` function exported.
if you still want to rename it, (it will be `executor.executor(...)`) i'll do.
4. except for the init part, i think we are pretty much a-like.
and if i'll get explanation why init part is needed i'll can add it here as well..

**Comment by DxCx** [comment]
reopen  @ Offical

**Pull request closed without merging**